docs/pktoken.md

```diff
@@ -0,0 +1,702 @@
+    "head_ref": "",
+    "base_ref": "",
+    "event_name": "push",
+    "ref_protected": "false",
+    "ref_type": "branch",
+    "workflow_ref": "openpubkey/gha-test/.github/workflows/test.yml@refs/heads/main",
+    "workflow_sha": "6b906a4153c61a2486973a1347db8300dc9ce3ee",
+    "job_workflow_ref": "openpubkey/gha-test/.github/workflows/test.yml@refs/heads/main",
+    "job_workflow_sha": "6b906a4153c61a2486973a1347db8300dc9ce3ee",
+    "runner_environment": "github-hosted",
+    "iss": "https://token.actions.githubusercontent.com",
+    "nbf": 1722185193,
+    "exp": 1722186093,
+    "iat": 1722185793}
+"signatures":[
+  {
+    "protected":{
+    "alg": "GQ256",
+    "jkt": "UJCvHZiaJcaQDc2tJbP_kgtgxB-OcKd1lwD76M3riUY",
+    "kid": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ikh5cTROQVRBanNucUM3bWRydEFoaHJDUjJfUSIsImtpZCI6IjFGMkFCODM0MDRDMDhFQzlFQTBCQjk5REFFRDAyMTg2QjA5MURCRjQifQ",
+    "typ": "JWT"},
+    "signature":"h4lbgJd... (5504 base64 characters)"
+  },
+  {
+    "protected": {
+    "alg": "ES256",
+    "rz": "bca0353ea63adbfce72032ab7d8fb7940def3488ca0765546a89d46760113c70",
+    "typ": "CIC",
+    "upk": {
+        "alg": "ES256",
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "5BP8B8bXgf0OFxHLJS5LSFlPOsfdIvf2tJU_3mwTGNE",
+        "y": "7KzWJi88qdZOI_j-kUG2aPjkzEA7IGMXFp1f-jdt28I"
+    }},
+    "signature":"5dfaFj..."
+  }
+]}
+```
+
+#### Audience-Commitment
+
+Machine identity ID Token issuance flows typically allow the identity requesting the ID Token to specify any value for the audience `aud`. Most machine identity flows do not use a `nonce`. To support OpenPubkey we use the `aud` in exactly the same way as the `nonce` commitment:
+
+```ascii
+nonce = SHA3(
+  BASE64URL({"alg": "ES256",
+    "rz": "bca0353ea63adbfce72032ab7d8fb7940def3488ca0765546a89d46760113c70",
+    "typ": "CIC",
+    "upk": {
+        "alg": "ES256",
+        "crv": "P-256",
+        "kty": "EC",
+        "x": "5BP8B8bXgf0OFxHLJS5LSFlPOsfdIvf2tJU_3mwTGNE",
+        "y": "7KzWJi88qdZOI_j-kUG2aPjkzEA7IGMXFp1f-jdt28I"
+    }}))
+```
+
+User identity ID Token issuance flows set the `aud` to a unique identifier to scope the ID Token to a particular service or OIDC client. This is done, among other reasons to prevent a malicious service from replaying the ID Tokens it receives from users to impersonate those users to a another service. Allowing the requesting party to specify the `aud` as is done in machine identity would be insecure for user identity. However it is both secure and the primary pattern in machine identity flows.
+
+#### GQ Signatures Are Required For Aud-Commitment PK Tokens
+
+The main use case of machine identity audience-commitment PK Tokens is to create publicly published signatures. As in such cases GQ signatures or ZK proofs should always be used, in such cases we require the use of GQ signatures. This eliminates the risks of accidental misconfigurations where a GQ signature should be used but is not used.
```

Comment:
Suggest rewording to remove the repetition: "As in such cases GQ signatures or ZK proofs should always be used, in such cases we require the use of GQ signatures." to "In such cases, GQ signatures or ZK proofs should always be used."

Reply:
Fixed